Realm/RLMResultsBase.h

```diff
@@ -0,0 +1,21 @@
+//
+//  RLMResultsBase.h
+//  Realm
+//
+//  Created by Adam Fish on 11/25/15.
+//  Copyright © 2015 Realm. All rights reserved.
+//
+
+#import <Foundation/Foundation.h>
+
+@class RLMResults;
+
+// A base class for Swift generic Results to make it possible to interact with
+// them from obj-c
+@interface RLMResultsBase : NSObject <NSFastEnumeration>
```

Comment:
This will cause `RLMResultsBase` to be part of the public API in both Objective-C and Swift, along with the two members declared here (`rlmResults` and `initWithResults:`). It's unfortunately necessary for `RLMListBase`, but avoidable here.